google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -445,27 +463,56 @@ public GrpcBlobReadChannel reader(String bucket, String blob, BlobSourceOption..
 
   @Override
   public GrpcBlobReadChannel reader(BlobId blob, BlobSourceOption... options) {
-    Object pb = codecs.blobId().encode(blob);
-    return new GrpcBlobReadChannel(grpcStorageStub.readObjectCallable(), () -> pb);
+    final Map<StorageRpc.Option, ?> optionsMap = StorageImpl.optionMap(options);
+    ReadObjectRequest request = getReadObjectRequest(blob, optionsMap);
+    return new GrpcBlobReadChannel(grpcStorageStub.readObjectCallable(), request);
   }
 
   @Override
   public void downloadTo(BlobId blob, Path path, BlobSourceOption... options) {
-    try (WritableByteChannel out = Files.newByteChannel(path, WRITE_OPS);
-        GrpcBlobReadChannel in = reader(blob, options)) {
-      ByteStreams.copy(in, out);
+
+    final Map<StorageRpc.Option, ?> optionsMap = StorageImpl.optionMap(options);
+
+    // TODO: handle StorageRpc.Option.RETURN_RAW_INPUT_STREAM impacts
+
+    ReadObjectRequest readObjectRequest = getReadObjectRequest(blob, optionsMap);
+    UnbufferedReadableByteChannelSession<Object> session =
+        ResumableMedia.gapic()
+            .read()
+            .byteChannel(grpcStorageStub.readObjectCallable())
+            .unbuffered()
+            .setReadObjectRequest(readObjectRequest)
+            .build();
+
+    try (UnbufferedReadableByteChannel r = session.open();
+        WritableByteChannel w = Files.newByteChannel(path, WRITE_OPS)) {
+      ByteStreams.copy(r, w);
     } catch (IOException e) {
-      throw new StorageException(e);
+      throw StorageException.coalesce(e);
     }
   }
 
   @Override
   public void downloadTo(BlobId blob, OutputStream outputStream, BlobSourceOption... options) {
-    try (WritableByteChannel out = Channels.newChannel(outputStream);
-        GrpcBlobReadChannel in = reader(blob, options)) {
-      ByteStreams.copy(in, out);
+
+    final Map<StorageRpc.Option, ?> optionsMap = StorageImpl.optionMap(options);
+
+    // TODO: handle StorageRpc.Option.RETURN_RAW_INPUT_STREAM impacts
```

Comment:
gRPC will always provide compressed data; maybe we can decompress data for the user to match up with the existing JSON API experience